dotnet/src/Connectors/Connectors.OpenAI.UnitTests/Services/OpenAIChatCompletionServiceTests.cs

```diff
@@ -1106,6 +1106,41 @@ public async Task ItCreatesCorrectFunctionToolCallsWhenUsingNoneFunctionChoiceBe
         Assert.Equal("none", optionsJson.GetProperty("tool_choice").ToString());
     }
 
+    [Fact]
+    public async Task ItCreatesCorrectFunctionToolCallsWhenUsingRequiredFunctionChoiceBehaviorAsync()
+    {
+        // Arrange
+        var kernel = new Kernel();
+        kernel.Plugins.AddFromFunctions("TimePlugin", [
+            KernelFunctionFactory.CreateFromMethod(() => { }, "Date"),
+            KernelFunctionFactory.CreateFromMethod(() => { }, "Now")
+        ]);
+
+        var chatCompletion = new OpenAIChatCompletionService(modelId: "gpt-3.5-turbo", apiKey: "NOKEY", httpClient: this._httpClient);
+
+        using var response = new HttpResponseMessage(HttpStatusCode.OK) { Content = new StringContent(File.ReadAllText("TestData/chat_completion_test_response.json")) };
+        this._messageHandlerStub.ResponseQueue.Enqueue(response);
+
+        var chatHistory = new ChatHistory();
+        chatHistory.AddUserMessage("Fake prompt");
```

Comment:
Curious when are we going to deprecate the `AddUserMessage` in favor of `Add(new)`
```suggestion
        chatHistory.AddUserMessage("Fake prompt");
```

Reply:
[Out of scope of this PR] We need to discuss if we really want to do that and prioritize if that is the case.